packages/chakra-components/src/components/Election/Election.tsx

```diff
@@ -91,17 +102,19 @@ export const useElectionProvider = ({
       setVoterAddress(address)
       const isIn = await client.isInCensus(election.id)
       setIsInCensus(isIn)
+      const censusType: CensusType = election.census.type
+      setCensusType(censusType)
 
       let left = 0
-      if (isIn) {
+      if (isIn || censusType == CensusType.WEIGHTED) {
         // no need to check votes left if member ain't in census
         left = await client.votesLeftCount(election.id)
         setVotesLeft(left)
 
         const voted = await client.hasAlreadyVoted(election.id)
         setVoted(voted)
       }
-      setIsAbleToVote(left > 0 && isIn)
+      setIsAbleToVote((left > 0 && isIn) || censusType == CensusType.CSP)
```

Comment:
Same here, unless it's intended (I doubt it for this case), use strict check always.